src/client/kube_store/index.ts

```diff
@@ -233,6 +81,20 @@ export default class KubeStore {
     return this.instance || (this.instance = new this(''));
   }
 
+  public patchResource(resource: KubeResource, name: string, patch: object): object {
+    const key = this.determineKey(resource, name);
+    const currentResource = this.data()[key][name];
+    const patchedObject = JsonMergePatch.apply(currentResource, patch);
+    const localStorageData = {...mocked_data, ...patchedObject};
```

Comment:
This is not creating the structure as you want.
This is taking 'patchedObject' and putting it at the root level of the object, instead of updating it in the nested key it needs to go in.